src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4012,20 +4013,20 @@ void setAnalyzedNullabilityAsContinuation(
             (int slot, NullableFlowState resultState, Func<TypeSymbol, MethodSymbol?, int>? completion) inferInitialObjectState(
                 BoundExpression node, TypeSymbol type, MethodSymbol? constructor,
                 ImmutableArray<BoundExpression> arguments, ImmutableArray<VisitResult> argumentResults,
-                bool isTargetTyped)
+                bool isTargetTyped,
+                bool hasObjectInitializer)
             {
                 if (isTargetTyped)
                 {
-                    return (-1, NullableFlowState.NotNull, inferInitialObjectStateAsContinuation(node, arguments, argumentResults));
+                    return (-1, NullableFlowState.NotNull, inferInitialObjectStateAsContinuation(node, arguments, argumentResults, hasObjectInitializer));
                 }
 
                 Debug.Assert(node.Kind is BoundKind.ObjectCreationExpression or BoundKind.DynamicObjectCreationExpression or BoundKind.NewT or BoundKind.NoPiaObjectCreationExpression);
 
-                var argumentTypes = argumentResults.SelectAsArray(ar => ar.RValueType);
-
                 int slot = -1;
                 var resultState = NullableFlowState.NotNull;
-                if (type is object)
+                if (type is object &&
+                    (hasObjectInitializer || type.IsStructType()))
                 {
                     slot = GetOrCreatePlaceholderSlot(node);
```

Comment:
A slot was created for each `new()` expression which resulted in a large `BitVector` for the `LocalState`, and since the `LocalState` is cloned before each statement, there were many copies created.

The fix is to avoid creating a slot for a `new()` expression unnecessarily. Specifically, we don't create a slot if the target type is a reference type and there is no object initializer, since there are no references to fields of the instance.

An alternative, to fix this more generally, we could consider freeing up and re-using slots when the corresponding expression or local is no longer in scope. That would be a much more significant change though.